services/web/server/src/simcore_service_webserver/meta_projects.py

```diff
@@ -0,0 +1,83 @@
+""" Access to the to projects module
+
+    - Adds a middleware to intercept /projects/* requests
+    - Implements a MetaProjectRunPolicy policy (see director_v2_abc.py) to define how meta-projects run
+
+"""
+
+
+import logging
+import re
+from typing import List, Tuple
+
+from aiohttp import web
+from aiohttp.web_middlewares import _Handler
+from models_library.basic_regex import UUID_RE
+from models_library.projects import ProjectID
+
+from ._meta import api_version_prefix as VTAG
+from .director_v2_api import AbstractProjectRunPolicy
+from .meta_iterations import get_or_create_runnable_projects, get_runnable_projects_ids
+from .meta_version_control import CommitID, VersionControlForMetaModeling
+from .projects.projects_handlers import RQ_REQUESTED_REPO_PROJECT_UUID_KEY
+
+log = logging.getLogger(__name__)
+
+
+# SEE https://github.com/ITISFoundation/osparc-simcore/blob/master/services/web/server/src/simcore_service_webserver/api/v0/openapi.yaml#L8563
+URL_PATTERN = re.compile(rf"^\/{VTAG}\/projects\/({UUID_RE})[\/]{{0,1}}")
+
+
+@web.middleware
+async def projects_redirection_middleware(request: web.Request, handler: _Handler):
+    """Intercepts /projects/{project_id}* requests and redirect them to the copy @HEAD"""
```

Comment:
I am not sure what his is doing. Could you provide an example, maybe it will make it easier to comprehend.